pkg/uefivars/boot/efiDppMedia.go

```diff
@@ -0,0 +1,258 @@
+)
+
+type EfiDppMediaSubType EfiDevPathProtoSubType
+
+const (
+	//DppTypeMedia, pg 319 +
+	DppMTypeHdd      EfiDppMediaSubType = iota + 1 //0x01
+	DppMTypeCd                                     //0x02
+	DppMTypeVendor                                 //0x03
+	DppMTypeFilePath                               //0x04 //p321
+	DppMTypeMedia                                  //0x05 //media protocol i.e. filesystem format??
+	DppMTypePIWGFF                                 //0x06
+	DppMTypePIWGFV                                 //0x07
+	DppMTypeRelOff                                 //0x08
+	DppMTypeRAM                                    //0x09
+)
+
+func (e EfiDppMediaSubType) String() string {
+	switch e {
+	case DppMTypeHdd:
+		return "HDD"
+	case DppMTypeCd:
+		return "CD"
+	case DppMTypeVendor:
+		return "Vendor"
+	case DppMTypeFilePath:
+		return "FilePath"
+	case DppMTypeMedia:
+		return "Media"
+	case DppMTypePIWGFF:
+		return "PIWG Firmware File"
+	case DppMTypePIWGFV:
+		return "PIWG Firmware Volume"
+	case DppMTypeRelOff:
+		return "Relative Offset"
+	case DppMTypeRAM:
+		return "RAMDisk"
+	default:
+		return fmt.Sprintf("UNKNOWN-0x%x", uint8(e))
+	}
+}
+
+//struct in EfiDevicePathProtocol for DppMTypeHdd
+type DppMediaHDD struct {
+	Hdr EfiDevicePathProtocolHdr
+
+	PartNum   uint32             //index into partition table for MBR or GPT; 0 indicates entire disk
+	PartStart uint64             //starting LBA. only used for MBR?
+	PartSize  uint64             //size in LB's. only used for MBR?
+	PartSig   uefivars.MixedGuid //format determined by SigType below. unused bytes must be 0x0.
+	PartFmt   uint8              //0x01 for MBR, 0x02 for GPT
+	SigType   uint8              //0x00 - none; 0x01 - 32bit MBR sig (@ 0x1b8); 0x02 - GUID
+}
+
+var _ EfiDevicePathProtocol = (*DppMediaHDD)(nil)
+
+func ParseDppMediaHdd(h EfiDevicePathProtocolHdr, b []byte) (*DppMediaHDD, error) {
+	if len(b) < 38 {
+		return nil, ErrParse
+	}
+	hdd := &DppMediaHDD{
+		Hdr:       h,
+		PartNum:   binary.LittleEndian.Uint32(b[:4]),
+		PartStart: binary.LittleEndian.Uint64(b[4:12]),
+		PartSize:  binary.LittleEndian.Uint64(b[12:20]),
+		//PartSig:   b[20:36], //cannot assign slice to array
+		PartFmt: b[36],
+		SigType: b[37],
+	}
+	copy(hdd.PartSig[:], b[20:36])
+	return hdd, nil
+}
+
+func (e *DppMediaHDD) Header() EfiDevicePathProtocolHdr { return e.Hdr }
+func (e *DppMediaHDD) ProtoSubTypeStr() string {
+	return EfiDppMediaSubType(e.Hdr.ProtoSubType).String()
+}
+func (e *DppMediaHDD) String() string {
+	//             (part#,pttype,guid,begin,length)
+	return fmt.Sprintf("HD(%d,%s,%s,0x%x,0x%x)", e.PartNum, e.pttype(), e.sig(), e.PartStart, e.PartSize)
+}
+
+func (e *DppMediaHDD) Resolver() (EfiPathSegmentResolver, error) {
+	allBlocks, err := storage.GetBlockStats()
+	if err != nil {
+		return nil, err
+	}
+	var blocks []storage.BlockDev
+	if e.SigType == 2 {
+		guid := e.PartSig.ToStdEnc().String()
+		blocks, err = storage.PartitionsByGUID(allBlocks, guid)
```

Comment:
PartitionsByGUID is actually totally wrong. I think it got merged as pkg/mount/block.BlockDevices.FilterByGUID, but the implementation is still wrong. Just FYI. I plan on fixing that if nobody else gets it first.